### ISSUE TITLE
Godot 4 - Can't access custom class properties and methods in variable inside Autoload script

### ISSUE THREAD

**Issue by mapedorr** [godotengine/godot#74058]
### Godot version

v4.0.rc6.official [0cd148313]

### System information

Windows 11

### Issue description

Having an Autoload script with a custom type (`subtype` in the following code chunk) defined by a constant preload, doesn't show the properties and methods of said custom type when typing code in the code editor *. The same scenario worked on Godot 3.

*Autocompletion actually works if the variable in the Autoload script is a Godot type (`coco` in the following code chunk)

**MyAutoload.gd**
```gdscript
const MySubtype := preload("res://MySubtype.gd")

var subtype: MySubtype : get = get_subtype
var coco := "Coco"

func get_subtype() -> MySubtype: return MySubtype.new()
```

https://user-images.githubusercontent.com/4536477/221595866-334c5c2b-70ca-442a-bdae-5e6224d14932.mov

### Steps to reproduce

The project in the ZIP has all the classes used to reproduce the issue. The code shown in the video is in the **_MyNode.gd_** script.

### Minimal reproduction project

[godot_4-autoload_issue.zip](https://github.com/godotengine/godot/files/10840926/godot_4-autoload_issue.zip)

I also attach a Godot 3 project with the same scenario. There you can see that autocompletion works when doing `MyAutoload.subtype.`.

[godot_3.5.1-autoload_working.zip](https://github.com/godotengine/godot/files/10841063/godot_3.5.1-autoload_working.zip)

**Comment by lyuma** [comment]
In earlier 4.0 builds, I used to workaround issues with autoloads in `_ready` by delaying some initialization by a frame.

Does it work if you do something like this?
```
func ready_next_frame():
    prints(MyAutoload....)
    ...

func _ready():
   call_deferred(self.ready_next_frame)
```

**Comment by mapedorr** [comment]
@lyuma No. The point is that I want to be able to access, while coding, the vars and methods in that `subtype` variable (of type `MySubtybe`) that is in **MyAutoload.gd**.

**Comment by Mickeon** [comment]
Happy to say that this seems to no longer be an issue in 4.3 dev, unless I'm mistaken? 
![image](https://github.com/godotengine/godot/assets/66727710/37d9aaca-dd65-4849-96b2-4945fa730dc5)
![image](https://github.com/godotengine/godot/assets/66727710/5d1695b1-ef32-4889-9ed4-a02ee0ef0b63)

**Comment by matheusmdx** [comment]
I can confirm this is already solved since v4.3.dev3.official [36e943b6b]

![image](https://github.com/godotengine/godot/assets/44306054/8bac36f3-6b4e-493d-9385-7f67a4e41d4d)
